src/Form/FormErrorNormalizer.php

```diff
@@ -14,34 +14,29 @@
 
 class FormErrorNormalizer implements NormalizerInterface
 {
-    protected TranslatorInterface $translator;
-    protected ConfigurationRepository $configurationRepository;
-
-    public function __construct(TranslatorInterface $translator, ConfigurationRepository $configurationRepository)
-    {
-        $this->translator = $translator;
-        $this->configurationRepository = $configurationRepository;
+    public function __construct(
+        protected readonly TranslatorInterface $translator,
+        protected readonly ConfigurationRepository $configurationRepository
+    ) {
     }
 
     /**
      * {@inheritDoc}
      */
-    public function normalize($object, $format = null, array $context = [])
+    public function normalize($object, ?string $format = null, array $context = [])
```

Comment:
I agree, but interesting how the function is declared in the interface

```php
public function normalize(mixed $object, string $format = null, array $context = []);
```

Reply:
I believe this has something to do with backwards compatibility (nullable types were [introduced in PHP 7.1](https://www.php.net/manual/en/migration71.new-features.php#migration71.new-features.nullable-types)) and since that's the case there's even a [style rule in PHP-CS-Fixer](https://cs.symfony.com/doc/rules/function_notation/nullable_type_declaration_for_default_null_value.html) about it 🤷🏽‍♂️